app/controllers/results_controller.rb

```diff
@@ -138,12 +138,11 @@ def edit
 
   def run_tests
     grouping_id = params[:grouping_id]
-    # TODO: The submission id is set incorrectly as the grouping_id
     submission_id = Result.find(params[:id]).submission.id
 
     begin
       AutomatedTestsHelper.request_a_test_run(grouping_id,
-                                              'submission',
+                                              'request',
```

Comment:
This is true, but I believe the student button doesn't use this method, but instead `execute_test_run` in `AutomatedTestsController`.

I looked into it more, and that method calls the `AutomatedTestsController`'s `run_tests` method, so there is a name collision, which may be contributing to the confusion.